### PR TITLE
Fixed Issue 1114

### PR DESCRIPTION
**Pull request by cmendible**
# Description

Secret API should not carry response when returning 204

According to HTTP guidelines returning 204 is not an error and should not carry a response.

So:
`respondWithError(c.RequestCtx, 204, NewErrorResponse("ERR_SECRET_NAME_NOT_FOUND", ""))`
should change to:
`respondEmpty(c.RequestCtx, 204)`

https://github.com/dapr/dapr/blob/55afe9dd2d5b1ba118c4d4c875bf873622f387cd/pkg/http/api.go#L409

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1114

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation
* [ ] Specification has been updated
* [ ] Provided sample for the feature

### CODE REVIEW

File: pkg/http/api.go [hunk 1]
```diff
@@ -406,7 +406,7 @@ func (a *api) onGetSecret(c *routing.Context) error {
 	}
 
 	if resp.Data == nil {
```
Comment (yaron2) [inline pkg/http/api.go]: Since `resp` is a pointer ref,  can this be `if resp == nil || resp.Data == nil`?

Comment (cmendible) [reply pkg/http/api.go]: In this case resp is not a pointer ref:

https://github.com/dapr/components-contrib/blob/d40be01831c423aea27e09933a64e175bb3d8575/secretstores/secret_store.go#L13

which is different approach from the one used in the StateStore:

https://github.com/dapr/components-contrib/blob/d40be01831c423aea27e09933a64e175bb3d8575/state/store.go#L13 

Should we update the SecretStore component?

Comment (yaron2) [reply pkg/http/api.go]: Yes

Comment (yaron2) [reply pkg/http/api.go]: No, right. If it's a non pointer is should be fine.

**Comment by yaron2** [comment]
@youngbupark can you please check why the build tests are failing here?

**Comment by youngbupark** [comment]
@yaron2 it is failing while cloning. We need to file ticket

**Comment by yaron2** [comment]
> @yaron2 it is failing while cloning. We need to file ticket

Based on e2e tests passing and the actual code change, do you think its safe to merge this?

**Comment by youngbupark** [comment]
> > @yaron2 it is failing while cloning. We need to file ticket
> 
> Based on e2e tests passing and the actual code change, do you think its safe to merge this?

build is working again.

**Pull request merged**